source/extensions/filters/http/common/jwks_fetcher.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "envoy/api/v2/core/http_uri.pb.h"
+#include "envoy/common/pure.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "jwt_verify_lib/jwks.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+
+class JwksFetcher {
+public:
+  typedef std::unique_ptr<JwksFetcher> JwksFetcherPtr;
+
+  class JwksReceiver {
+  public:
+    enum class Failure {
+      unknown,
+      network,
+      invalid_jwks,
+    };
+
+    virtual ~JwksReceiver(){};
+    /*
+     * Successful retrieval callback.
+     * of the returned JWKS object.
+     * @param jwks the JWKS object retrieved.
+     */
+    virtual void onJwksSuccess(google::jwt_verify::JwksPtr&& jwks) PURE;
+    /*
+     * Retrieval error callback.
+     * * @param reason the failure reason.
+     */
+    virtual void onJwksError(Failure reason) PURE;
+  };
+
+  virtual ~JwksFetcher(){};
+
+  /*
+   * Close/stop any inflight request.
+   */
+  virtual void close() PURE;
```

Comment:
rename it to cancel?

Reply:
Changed to cancel() in e6b6b0 as per review comment.